lib/context_request_middleware/request/cookie_session_id_retriever.rb

```diff
@@ -1,21 +1,13 @@
 # frozen_string_literal: true
 
+require 'context_request_middleware/request/base'
+
 module ContextRequestMiddleware
   module Request
     # Class for retrieving the session if set via rack cookie.
     # This requires the session id to be stored in '_session_id'
     # cookie key.
-    class CookieSessionIdRetriever
-      include ActiveSupport::Configurable
-
-      def initialize(request)
-        @request = request
-      end
-
-      def call
-        Rack::Utils.parse_cookies(@request.env)['_session_id'] ||
-          (@request.env['action_dispatch.cookies'] || {})['_session_id']
-      end
+    class CookieSessionIdRetriever < Base
```

Comment:
I don't think you necessarily need the 'Base' because you only use it ones. And the other retriever from above makes it different ;) .

Reply:
The reason I added a Base class is that I was thinking in the near future 
more classes will be added probably to `lib/context_request_middleware/request/*`
as this is a configurable gem as I see from here 
```ruby
config_accessor(:request_context_retriever, instance_accessor: false) do
    'cookie_session_id_retriever'
end
```
Anyway, I will remove the base and return the old version regarding your comment.